index.d.ts

```diff
@@ -788,10 +783,191 @@ type ExecaCommonError = {
 	This is `undefined` unless the subprocess exited due to an `error` event or a timeout.
```

Comment:
I would invert the formulation:

```suggestion
	This exists only if the subprocess exited due to an `error` event or a timeout.
```

Also because technically the property is not `undefined`, but rather non-existent. TypeScript can differentiate between those for users with the `--exactOptionalPropertyTypes` flag.